libjwt/jwt.c

```diff
@@ -116,23 +116,27 @@ int jwt_set_alg(jwt_t *jwt, jwt_alg_t alg, unsigned char *key, int len)
 	case JWT_ALG_HS256:
 	case JWT_ALG_HS384:
 	case JWT_ALG_HS512:
-		if (!key || len != key_len)
+		if (!key)
 			return EINVAL;
+		// key length is not required to equal jwt_alg_key_len (msg digest len)
+		// HMAC algorithm will either pad or truncate the key as necessary.
+		// if (!key || len != key_len)
+		// 	return EINVAL;
```

Comment:
Same here. Also, use /* comment */ styles, even on multi-line (see other parts of the code for examples).